docs/DeveloperGuide.adoc

```diff
@@ -767,6 +785,255 @@ Use case ends.
 +
 Use case resumes at step 2.
 
+[discrete]
+=== Use case: Add person details
+
+*MSS*
+
+1.  User requests to create a contact with more details including birthday, home call number and so on
+2.  User types in these details
+3.  AddressBook adds a new contact with these details
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. User types with wrong format.
++
+[none]
+** 2a1. AddressBook shows an error message.
++
+Use case ends.
+
+[discrete]
+=== Use case: Use right click menu
+
+*MSS*
+
+1.  User requests to delete or edit one person
+2.  User right clicks that contact
+3.  AddressBook displays a menu with delete and edit options
+4.  User choose delete option
+5.  AddressBook delete that contact
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 4a. User choose edit option.
++
+[none]
+** 4a1. AddressBook shows a dialogue for editing.
+** 4a2. User types in information.
+** 4a3. AddressBook saves these information.
++
+Use case ends.
+
+[discrete]
+=== Use case: Show contact according to categories of labels
+
+*MSS*
+
+1.  User requests to show all persons with a specific label
+2.  AddressBook shows a list a person
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The label does not exist.
++
+[none]
+** 1a1. AddressBook shows a empty list.
++
+Use case ends.
+
+[discrete]
+=== Use case: Delete confirmation
+
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User requests to delete a specific person in the list
+4.  AddressBook displays a dialogue window asking user to confirm deletion
+5.  User confirms deletion request
+6.  AddressBook deletes the person
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The given index is invalid.
++
+[none]
+** 1a1. AddressBook shows an error message.
++
+Use case ends.
+
+[none]
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[none]
+* 5a. User cancels deletion.
++
+Use case ends.
+
+[discrete]
+=== Use case: Flexible contact addition
+
+*MSS*
+
+1.  User requests to add a person
+2.  User inputs desired details, eg: address, email, birthday, etc...
```

Comment:
Step 1 and 2 can be combined into `User requests to add a person with some missing attributes`